### PR TITLE
[Chipmunk] build via cmake

### PR DESCRIPTION
**Pull request by comedinha**
Actually the chipmunk only can build via msvc, this change will allow build via cmake.

- What does your PR fix? Fixes #
https://github.com/microsoft/vcpkg/issues/10802

- Which triplets are supported/not supported? Have you updated the CI baseline?
I tested on windows x64 and mingw-w64-static.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/chipmunk/portfile.cmake [hunk 1]
```diff
@@ -1,22 +1,4 @@
...
-if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
-    set(CHIPMUNK_CONFIGURATION_SUFFIX " DLL")
-else()
-    if (VCPKG_CRT_LINKAGE STREQUAL dynamic)
-        set(CHIPMUNK_CONFIGURATION_SUFFIX "")
-    else()
-        set(CHIPMUNK_CONFIGURATION_SUFFIX " SCRT")
-    endif()
-endif()
+include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/chipmunk/portfile.cmake:1]:
```suggestion
```
This function is no longer needed here.

File: ports/chipmunk/portfile.cmake [hunk 2]
```diff
@@ -26,43 +8,96 @@ vcpkg_from_github(
...
+    string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" KEYSTONE_BUILD_STATIC)
+    string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" KEYSTONE_BUILD_SHARED)
+
+    vcpkg_configure_cmake(
+        SOURCE_PATH ${SOURCE_PATH}
+        PREFER_NINJA
+        OPTIONS -DBUILD_DEMOS=OFF
+                -DBUILD_SHARED=KEYSTONE_BUILD_SHARED
+                -DBUILD_STATIC=KEYSTONE_BUILD_STATIC
+                -DINSTALL_STATIC=KEYSTONE_BUILD_STATIC
```
Comment (NancyLi1013) [inline ports/chipmunk/portfile.cmake:79]:
```suggestion
        OPTIONS 
            -DBUILD_DEMOS=OFF
            -DBUILD_SHARED=${KEYSTONE_BUILD_SHARED}
            -DBUILD_STATIC=${KEYSTONE_BUILD_STATIC}
            -DINSTALL_STATIC=${KEYSTONE_BUILD_STATIC}
```

File: ports/chipmunk/portfile.cmake [hunk 3]
```diff
@@ -26,43 +8,96 @@ vcpkg_from_github(
...
-    "${SOURCE_PATH}/msvc/VS2015/chipmunk/${CHIPMUNK_ARCH}/Debug${CHIPMUNK_CONFIGURATION_SUFFIX}/chipmunk.lib"
-    DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib
-)
-file(INSTALL
-    "${SOURCE_PATH}/msvc/VS2015/chipmunk/${CHIPMUNK_ARCH}/Release${CHIPMUNK_CONFIGURATION_SUFFIX}/chipmunk.lib"
-    DESTINATION ${CURRENT_PACKAGES_DIR}/lib
-)
 file(INSTALL
     ${SOURCE_PATH}/include/chipmunk
     DESTINATION ${CURRENT_PACKAGES_DIR}/include
```
Comment (NancyLi1013) [inline ports/chipmunk/portfile.cmake:103]: Could you please update 

`file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/chipmunk RENAME copyright)`

as

`file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`

?

Comment (NancyLi1013) [reply ports/chipmunk/portfile.cmake:103]: `message(STATUS "Installing done")`

This line should be in msbuild part.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]
You might try to remove msbuild part and verify if it can build fine on Windows via CMake.

**Comment by NancyLi1013** [comment]
Please also update version in `CONTROl `file.

**Comment by comedinha** [comment]
@NancyLi1013 changes made.

**Comment by NancyLi1013** [comment]
> You might try to remove msbuild part and verify if it can build fine on Windows via CMake.

Have you tried to build via CMake on all platforms?

**Comment by comedinha** [comment]
> > You might try to remove msbuild part and verify if it can build fine on Windows via CMake.
> 
> Have you tried to build via CMake on all platforms?

Yeah, tested and fixed for x64 UWP

File: ports/chipmunk/portfile.cmake [hunk 4]
```diff
@@ -26,48 +6,33 @@ vcpkg_from_github(
...
-    "${SOURCE_PATH}/msvc/VS2015/chipmunk/${CHIPMUNK_ARCH}/Debug${CHIPMUNK_CONFIGURATION_SUFFIX}/chipmunk.lib"
-    DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib
-)
-file(INSTALL
-    "${SOURCE_PATH}/msvc/VS2015/chipmunk/${CHIPMUNK_ARCH}/Release${CHIPMUNK_CONFIGURATION_SUFFIX}/chipmunk.lib"
-    DESTINATION ${CURRENT_PACKAGES_DIR}/lib
-)
-file(INSTALL
-    ${SOURCE_PATH}/include/chipmunk
+    ${SOURCE_PATH}/include/${PORT}
```
Comment (NancyLi1013) [inline ports/chipmunk/portfile.cmake:34]:
```suggestion
    ${SOURCE_PATH}/include/chipmunk
```
This should be kept consistent with source code.

Generally, we recommend to use ${PORT} in these cases:
1) Install license to share/${PORT}.
2) Install some documents to share/${PORT}.
3) Other files that need to be listed to share/${PORT}.

File: scripts/ci.baseline.txt [hunk 5]
```diff
@@ -102,9 +102,6 @@ chartdir:x64-uwp=fail
 chartdir:x64-windows-static=fail
 chipmunk:arm64-windows=fail
 chipmunk:arm-uwp=fail
```
Comment (NancyLi1013) [inline scripts/ci.baseline.txt:104]: Could you please help confirm if chipmunk supports `arm`? Or not, we can remove these codes here and add `Supports `field to CONTROL file. Also need to add unsupported message in portfile.cmake.
`vcpkg_fail_port_install(ON_ARCH "arm")`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: scripts/ci.baseline.txt (hunk 5 above)
Comment (comedinha) [reply scripts/ci.baseline.txt:104]: Reading the doc of Chipmunk2D, arm is supported. But I can't test on my machine.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTm now, thanks for your PR @comedinha.

**Comment by BillyONeal** [comment]
Awesome, thanks for your help!

**Pull request merged**